internal/app/networkd/pkg/networkd/networkd.go

```diff
@@ -78,35 +79,47 @@ func (n *Networkd) Configure(ifaces ...*nic.NetworkInterface) error {
 	var (
 		err       error
 		resolvers []net.IP
+		mu        sync.Mutex
 	)
 
-	for _, iface := range ifaces {
-		// Bring up the interface
-		if err = n.Conn.LinkUp(&net.Interface{Index: int(iface.Index)}); err != nil {
-			log.Printf("failed to bring up %s: %v", iface.Name, err)
-			continue
-		}
+	var wg sync.WaitGroup
 
-		// Generate rtnetlink.AddressMessage for each address method defined on
-		// the interface
-		for _, method := range iface.AddressMethod {
-			log.Printf("configuring %s addressing for %s\n", method.Name(), iface.Name)
+	wg.Add(len(ifaces))
 
-			if err = n.configureInterface(method); err != nil {
-				// Treat as non fatal error when failing to configure an interface
-				log.Println(err)
-				continue
+	for _, iface := range ifaces {
+		go func(i *nic.NetworkInterface) {
+			defer wg.Done()
+			// Bring up the interface
+			if err = n.Conn.LinkUp(&net.Interface{Index: int(i.Index)}); err != nil {
+				log.Printf("failed to bring up %s: %v", i.Name, err)
+				return
 			}
 
-			if !method.Valid() {
-				continue
-			}
+			// Generate rtnetlink.AddressMessage for each address method defined on
+			// the interface
+			for _, method := range i.AddressMethod {
+				log.Printf("configuring %s addressing for %s\n", method.Name(), i.Name)
 
-			// Aggregate a list of DNS servers/resolvers
-			resolvers = append(resolvers, method.Resolvers()...)
-		}
+				if err = n.configureInterface(method); err != nil {
+					// Treat as non fatal error when failing to configure an interface
+					log.Println(err)
+					return
+				}
+
+				if !method.Valid() {
+					return
+				}
+
+				// Aggregate a list of DNS servers/resolvers
+				mu.Lock()
```

Comment:
unsure if it's better to use a lock here or just kick off another loop afterwards
```
for _, iface := range ifaces {
  go func(i *nic.NetworkInterface) { ... }
}
wg.Wait()
var resolvers []string
for _, iface := range ifaces {
  resolvers = append(resolvers, method.Resolvers()...)
}

Reply:
One for loop better than two maybe? IDK. We could probably go either way but int the end one doesn't have something over the other IMO.